aiida/cmdline/commands/cmd_node.py

```diff
@@ -61,6 +64,52 @@ def repo_ls(node, relative_path, color):
         echo.echo_critical(exception)
 
 
+@verdi_node_repo.command('dump')
+@arguments.NODE()
+@click.argument(
+    'output_directory',
+    type=click.Path(),
+    required=True,
+)
+@with_dbenv()
+def repo_dump(node, output_directory):
+    """Copy the repository files of a node to an output directory."""
+    from aiida.orm.utils.repository import FileType
+
+    output_directory = pathlib.Path(output_directory)
+
+    try:
+        output_directory.mkdir(parents=True, exist_ok=False)
+    except FileExistsError:
+        click.echo('Error: Invalid value for "OUTPUT_DIRECTORY": Path "{}" exists.'.format(output_directory))
+        raise click.Abort
+
+    def _copy_tree(key, output_dir):  # pylint: disable=too-many-branches
+        """
+        Recursively copy the content at the ``key`` path in the given node to
+        the ``output_dir``.
+        """
+        for file in node.list_objects(key=key):
+            # Not using os.path.join here, because this is the "path"
+            # in the AiiDA node, not an actual OS - level path.
+            file_key = file.name if not key else key + '/' + file.name
```

Comment:
This is a good point and I now start to wonder whether we actually handle this correctly and consistently everywhere in the repositiory interface. Anyhow, this is good for now.

Reply:
Well, that's only going to be a problem if ``os.sep`` is a backslash.. not the trickiest part for Windows support, I suppose 😄